lib/Driver/Compilation.cpp

```diff
@@ -421,14 +424,26 @@ int Compilation::performJobsImpl() {
   }
 
   int Result = EXIT_SUCCESS;
+  llvm::TimerGroup DriverTimerGroup("Driver Time Compilation");
+  std::map<std::string, std::unique_ptr<llvm::Timer>> DriverTimers;
```

Comment:
Given how the code is written, couldn't use an RAII type to hide all of this? I am imagining an API like:

// We want to make sure map is not allocating memory here unless we have ShowDriverTimeCompilation actually enabled. So I imagine you want something like SharedTimer has.
SimpleTimerGroup Group;

llvm::SmallString<128> CmdLine;
auto taskBegin = ... {
  if (ShowDriverTimeCompilation) {
    BeginCmd->printCommandLine(CmdLine);
    Group.startTimer(CmdLine)
  }
}

...